# Aufgabe 1.b)

In [1]:
def fixpunktiteration(f, n, x_0):
    x_i = x_0
    for i in range(n):
        fx_i = f(x_i)
        print(f"x_{i + 1} = f({x_i}) = {fx_i}")
        x_i = fx_i
    return x_i

In [2]:
f1 = lambda x: 5 + x - x**2
f2 = lambda x: 5/x
f3 = lambda x: 1 + x - (x**2)/5
f4 = lambda x: 0.5 * (x + 5/x)

all_f = [f1, f2, f3, f4]
for i, f in enumerate(all_f):
    print(f"now doing f_{i+1}...")
    fixpunktiteration(f, 6, 2.5)


now doing f_1...
x_1 = f(2.5) = 1.25
x_2 = f(1.25) = 4.6875
x_3 = f(4.6875) = -12.28515625
x_4 = f(-12.28515625) = -158.21022033691406
x_5 = f(-158.21022033691406) = -25183.68403939181
x_6 = f(-25183.68403939181) = -634243120.4799571
now doing f_2...
x_1 = f(2.5) = 2.0
x_2 = f(2.0) = 2.5
x_3 = f(2.5) = 2.0
x_4 = f(2.0) = 2.5
x_5 = f(2.5) = 2.0
x_6 = f(2.0) = 2.5
now doing f_3...
x_1 = f(2.5) = 2.25
x_2 = f(2.25) = 2.2375
x_3 = f(2.2375) = 2.23621875
x_4 = f(2.23621875) = 2.2360838904296876
x_5 = f(2.2360838904296876) = 2.2360696574218544
x_6 = f(2.2360696574218544) = 2.2360681548533163
now doing f_4...
x_1 = f(2.5) = 2.25
x_2 = f(2.25) = 2.236111111111111
x_3 = f(2.236111111111111) = 2.2360679779158037
x_4 = f(2.2360679779158037) = 2.23606797749979
x_5 = f(2.23606797749979) = 2.23606797749979
x_6 = f(2.23606797749979) = 2.23606797749979


# OUTPUT

**For $\Phi_{1}$**

$x^{(1)} = \Phi_{1}(2.5) = 1.25$

$x^{(2)} = \Phi_{1}(1.25) = 4.6875$

$x^{(3)} = \Phi_{1}(4.6875) = -12.28515625$

$x^{(4)} = \Phi_{1}(-12.28515625) = -158.21022033691406$

$x^{(5)} = \Phi_{1}(-158.21022033691406) = -25183.68403939181$

$x^{(6)} = \Phi_{1}(-25183.68403939181) = -634243120.4799571$


**For $\Phi_{2}$**

$x^{(1)} = \Phi_{2}(2.5) = 2.0$

$x^{(2)} = \Phi_{2}(2.0) = 2.5$

$x^{(3)} = \Phi_{2}(2.5) = 2.0$

$x^{(4)} = \Phi_{2}(2.0) = 2.5$

$x^{(5)} = \Phi_{2}(2.5) = 2.0$

$x^{(6)} = \Phi_{2}(2.0) = 2.5$


**For $\Phi_{3}$**

$x^{(1)} = \Phi_{3}(2.5) = 2.25$

$x^{(2)} = \Phi_{3}(2.25) = 2.2375$

$x^{(3)} = \Phi_{3}(2.2375) = 2.23621875$

$x^{(4)} = \Phi_{3}(2.23621875) = 2.2360838904296876$

$x^{(5)} = \Phi_{3}(2.2360838904296876) = 2.2360696574218544$

$x^{(6)} = \Phi_{3}(2.2360696574218544) = 2.2360681548533163$


**For $\Phi_{4}$**

$x^{(1)} = \Phi_{4}(2.5) = 2.25$

$x^{(2)} = \Phi_{4}(2.25) = 2.236111111111111$

$x^{(3)} = \Phi_{4}(2.236111111111111) = 2.2360679779158037$

$x^{(4)} = \Phi_{4}(2.2360679779158037) = 2.23606797749979$

$x^{(5)} = \Phi_{4}(2.23606797749979) = 2.23606797749979$

$x^{(6)} = \Phi_{4}(2.23606797749979) = 2.2360679774997$